### PR TITLE
feat: add brightscript server

### PR DESCRIPTION
**Pull request by glepnir**
close #2390

### CODE REVIEW

File: lua/lspconfig/server_configurations/bright_script.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
+local bin_name = 'bsc'
+local cmd = { bin_name, '--lsp', '--stdio' }
+local util = require "lspconfig/util"
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/bright_script.lua:3]:
```suggestion
local util = require 'lspconfig/util'
```

**Comment by casonadams** [comment]
Thanks for jamming this one through!  Sorry I didn't get the docs bit correct.

**Comment by glepnir** [comment]
don't worry it's ok

File: lua/lspconfig/server_configurations/bright_script.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    description = [[
+https://github.com/RokuCommunity/brighterscript
+
+`brightscript` can be installed via `npm`:
+```sh
+npm install -g brighterscript
+```
+]],
+    default_config = {
+      root_dir = [[root_pattern(".git")]],
```
Comment (casonadams) [inline lua/lspconfig/server_configurations/bright_script.lua:26]: Is the `"` here going to break tests too?

Comment (glepnir) [reply lua/lspconfig/server_configurations/bright_script.lua:26]: it's in `[[ ]]` so lint will  be pass.

**Comment by casonadams** [comment]
Very happy to get this change in here, I have some brightscripting to do tomorrow :smile:

**Comment by casonadams** [comment]
There is a broken link in the markdown doc page for `bright_script` now.

**Pull request merged**